src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs

```diff
@@ -35,8 +35,14 @@ public class SqlServerDataStoreConfiguration
         public SqlServerSchemaOptions SchemaOptions { get; set; } = new SqlServerSchemaOptions();
 
         /// <summary>
-        /// If set, Instructs the serivce to terminate when its schema reaches the specified version.
+        /// If set, Instructs the service to terminate when its schema reaches the specified version.
         /// </summary>
         public int? TerminateWhenSchemaVersionUpdatedTo { get; set; }
+
+        /// <summary>
+        /// true: registers ProcessTerminator
+        /// false: registers NoOpProcessTerminator
+        /// </summary>
+        public bool RegisterProcessTerminator { get; set; }
```

Comment:
>RegisterProcessTerminator [](http://example.com/codeflow?start=20&length=25)

wouldn't it be simplier to check TerminateWhenSchemaVersionUpdatedTo and register Noop if it's null, and normal one in other cases?

It feels weird to have two settings. It's makes posible to have noop and schema to be not null, and do nothing.

Reply:
Agreed, removed the new setting.